src/pages/TokenDetails/index.tsx

```diff
@@ -14,38 +14,41 @@ export const pageTimePeriodAtom = atomWithStorage<TimePeriod>('tokenDetailsTimeP
 export default function TokenDetailsPage() {
   const { tokenAddress, chainName } = useParams<{ tokenAddress?: string; chainName?: string }>()
   const chain = validateUrlChainParam(chainName)
-  const pageChainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const isNative = tokenAddress === NATIVE_CHAIN_ID
   const [timePeriod, setTimePeriod] = useAtom(pageTimePeriodAtom)
-  const [contract, duration] = useMemo(
-    () => [
-      { address: isNative ? nativeOnChain(pageChainId).wrapped.address : tokenAddress ?? '', chain },
-      toHistoryDuration(timePeriod),
-    ],
-    [chain, isNative, pageChainId, timePeriod, tokenAddress]
+  const [address, duration] = useMemo(
+    () => [isNative ? getNativeTokenDBAddress(chain) : tokenAddress ?? '', toHistoryDuration(timePeriod)],
+    [chain, isNative, timePeriod, tokenAddress]
   )
 
-  const { data: tokenQuery, loading: tokenQueryLoading } = useTokenQuery({
+  const { data: tokenQuery } = useTokenQuery({
     variables: {
-      contract: isNative ? { address: getNativeTokenDBAddress(chain), chain } : contract,
+      address,
+      chain,
     },
   })
 
   const { data: tokenPriceQuery } = useTokenPriceQuery({
     variables: {
-      contract,
+      address,
+      chain,
       duration,
     },
   })
 
-  if (!tokenQuery || tokenQueryLoading) return <TokenDetailsPageSkeleton />
+  const [currentPriceData, setCurrentPriceData] = useState(tokenPriceQuery)
+  useEffect(() => {
+    if (tokenPriceQuery) setCurrentPriceData(tokenPriceQuery)
+  }, [setCurrentPriceData, tokenPriceQuery])
```

Comment:
is this needed if the fetch function is memoized?

Reply:
When the timePeriod changes, tokenPriceQuery becomes null; I added this extra currentPriceData variable so that in the chart, we display existing data until new data  is available, rather than repeatedly showing a loading state